File: Project/Remix_pwa/app/Models/interfaces/HtmlElement.ts
import { MyAnimation } from "../implementations/Animation";

export abstract class HtmlElement {
    _animation: MyAnimation | null = null;
    _style: Map<string, string> = new Map<string, string>();
    _attributes: Map<string, string> = new Map<string, string>();

    get Animation(): MyAnimation | null {
        return this._animation;
    }

    set Animation(animation: MyAnimation | null) {
        this._animation = animation;
    }

    get Parameters(): Map<string, string> {
        return this._attributes;
    }

    get Style(): Map<string, string> {
        return this._style;
    }

    abstract get Name(): string;

    constructor() {
        this._style.set("left", "0px");
        this._style.set("top", "0px");
        this._style.set("position", "absolute");
        this._style.set("z-index", "0");
        this._style.set("animation-duration", "1s");
    }

    get ToString(): string {
        return `Html element '${this.Name}'`;
    }
}

File: Project/Blazor_pwa/Models/Interfaces/HtmlElement.cs
using Blazor_pwa.Models.Emums;
using Blazor_pwa.Models.Enums;
using Blazor_pwa.Models.Implementations;

namespace Blazor_pwa.Models.Interfaces
{
    public class HtmlElement
    {
        /// <summary>
        /// Type of the HTML element
        /// </summary>
        public HtmlElementType Type { get; private set; }

        public Dictionary<string, string> Attributes { get; private set; } = new();
        public Dictionary<string, string> Style { get; private set; } = new();
        public Dictionary<string, string> AnimationStyle { get; private set; } = new();
    }
}